plugins/serializers/splunkmetric/splunkmetric.go

```diff
@@ -0,0 +1,135 @@
+package splunkmetric
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/influxdata/telegraf"
+)
+
+type serializer struct {
+	TimestampUnits time.Duration
+}
+
+func NewSerializer(timestampUnits time.Duration) (*serializer, error) {
+	s := &serializer{
+		TimestampUnits: truncateDuration(timestampUnits),
+	}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	var serialized string
+
+	m, err := s.createObject(metric)
+	if err == nil {
+		serialized = m + "\n"
+	}
+
+	return []byte(serialized), nil
+}
+
+func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+
+	var serialized string
+
+	var objects []string
+
+	for _, metric := range metrics {
+		m, err := s.createObject(metric)
+		if err == nil {
+			objects = append(objects, m)
+		}
+	}
+
+	for _, m := range objects {
+		serialized = serialized + m + "\n"
+	}
+
+	return []byte(serialized), nil
+}
+
+func (s *serializer) createObject(metric telegraf.Metric) (metricString string, err error) {
+
+	/* Splunk supports one metric per line and has the following required names:
```

Comment:
This is not entirely accurate. Splunk's http event collector is not line-oriented. It's object oriented. You can shove multiple JSON objects on a single line and it's happy to consume them.
See this example: https://docs.splunk.com/Documentation/Splunk/7.1.1/Data/HTTPEventCollectortokenmanagement#Send_multiple_events_to_HEC_in_one_request
```
{"event": "Pony 1 has left the barn"}{"event": "Pony 2 has left the barn"}{"event": "Pony 3 has left the barn", "nested": {"key1": "value1"}}
```

Reply:
I agree, it's not line oriented, but it doesn't support reading an array of JSON objects as metrics. This is OK:
```javascript
{"event": "Pony 1 has left the barn"}{"event": "Pony 2 has left the barn"}{"event": "Pony 3 has left the barn", "nested": {"key1": "value1"}}
```

This is not OK:
```javascript
[{"event": "Pony 1 has left the barn"}{"event": "Pony 2 has left the barn"}{"event": "Pony 3 has left the barn", "nested": {"key1": "value1"}}]
```

Reply:
My point was that developer documentation should be accurate. We shouldn't tell the developers it's "one metric per line" if it's not.